sklearn/neighbors/lof.py

```diff
@@ -169,6 +177,14 @@ def fit_predict(self, X, y=None):
             Returns -1 for anomalies/outliers and 1 for inliers.
         """
 
+        # As fit_predict would be different from fit.predict, fit_predict is
+        # only available for outlier detection (novelty=False)
+
+        if self.novelty:
+            msg = ('fit_predict is not available when novelty=True. Use '
+                   'novelty=False if you want to predict on the training set.')
+            raise NotImplementedError(msg)
```

Comment:
I don't think this is the right error type. It makes it seem like one day we will implement this case

Reply:
A TypeError would be more normal for a bad call, while a ValueError would be possible too. None is quite satisfying.

Reply:
I would use ValueError